File: project/app/src/main/java/org/owntracks/android/injection/modules/android/ActivityModules/MapActivityModule.java
package org.owntracks.android.injection.modules.android.ActivityModules;

import androidx.appcompat.app.AppCompatActivity;

import org.owntracks.android.injection.scopes.PerActivity;
import org.owntracks.android.ui.map.MapActivity;
import org.owntracks.android.ui.map.MapMvvm;
import org.owntracks.android.ui.map.MapViewModel;

import dagger.Binds;
import dagger.Module;

@Module(includes = BaseActivityModule.class)
public abstract class MapActivityModule {

    @Binds
    @PerActivity
    abstract AppCompatActivity bindActivity(MapActivity a);

    @Binds abstract MapMvvm.ViewModel bindViewModel(MapViewModel viewModel);
}

File: project/app/src/main/java/org/owntracks/android/injection/modules/android/ActivityModules/MapActivityModule.kt
package org.owntracks.android.injection.modules.android.ActivityModules

import androidx.appcompat.app.AppCompatActivity
import dagger.Binds
import dagger.Module
import org.owntracks.android.injection.scopes.PerActivity
import org.owntracks.android.ui.map.MapActivity
import org.owntracks.android.ui.map.MapMvvm
import org.owntracks.android.ui.map.MapViewModel

@Module
abstract class MapActivityModule {
    @Binds
    @PerActivity
    abstract fun bindActivity(a: MapActivity?): AppCompatActivity?

    @Binds
    abstract fun bindViewModel(viewModel: MapViewModel): MapMvvm.ViewModel<MapMvvm.View>
}